### DISCUSSION TITLE
[Q&A] Deploy Glpi agent using GPO

### DISCUSSION THREAD

**Discussion by Chico0008** [glpi-project/glpi-agent#330] [Q&A]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

Hi
I want to deploy glpi agent with Gpo runned at startup of computer

so, on my AD server, i set a GPO to launch a .bat script on our network
\\s-fichiers\glpi\launcher.bat
the folder is shared and readable by everyone.

bat : if i got no glpi installation folder, i do the install
copy both glpi-agent-deployement.vbs, and glpi-agent.msi so c:\windows\temp
then run the vbs.

the vbs has been edited to take the msi file, and setuplocation set to c:\windows\temp

when i start my computer, then login, i can see the 2 files in my c:\windows\temp
but, installation do not seems to work
my mouse cursor keep switching beetween standard and task running, but agent is never installed.

i can't figure why it doesn't work :(

my launcher.bat
```
@echo off

REM test si le dossier d'install de l'agent GLPI est present, si oui, on sort sans rien faire
if exist "C:\Program Files\GLPI-Agent" goto end

REM si pas présent, on fait l'install
copy \\s-fichiers\glpi\glpi* c:\windows\temp
echo "Installation agent GLPI"
cscript c:\windows\temp\glpi-agent-deploy.vbs
del /f c:\windows\temp\glpi*

:end
exit
```

in VBS
```
SetupLocation = "c:\windows\temp"
Setup = "GLPI-Agent-1.5-git75728309-x64.msi"
```

the GPO
![image](https://user-images.githubusercontent.com/58886243/216288327-21f3456a-2f5b-4097-9342-020caddc5e47.png)

**Comment by g-bougard** [comment]
Hi @Chico0008 
you forgot to share your changed `SetupOptions`. People often makes mistakes with them as the syntax is totally different than the one for FI windows installer.

**Reply by Chico0008** [reply]
`SetupOptions = "/quiet RUNNOW=1 SERVER='http://glpi.viry.net/plugins/glpiinventory/' NO_SSL_CHECK=1 TASK_FREQUENCY=daily TASK_DAILY_MODIFIER=2"`

tried replacing /quiet with /qn or /passive, but nothing change.

**Reply by g-bougard** [reply]
This seems okay. Just a point, you don't need `NO_SSL_CHECK` if your url is not an `https://` one ;-)
And if indeed want to use an `hhtps://`, you should prefer to use the new `SSL_FINGERPRINT` option so you really can trust your server certificate.

Another point, you may not want to check if `C:\Program Files\GLPI-Agent` folder exists but looks for `HKEY_LOCAL_MACHINE\SOFTWARE\GLPI-Agent\Installer\Version` registry key. This is still what the VBS is doing to check if requested version is still installed or not.

For the problem, you may check the system eventviewer for any error related to MSI install.

You may also add `/L*V C:\\Windows\Temp\agent-install.log` just after `/quiet` to have a full msiexec run log.

Not being a GPO usage expert, does the script run with administrator privileges ?

**Reply by Chico0008** [reply]
Runing script at startup like this are executed as local system (admin)

here's the log
```
=== Verbose logging started: 02/02/2023  16:51:33  Build type: SHIP UNICODE 5.00.10011.00  Calling process: C:\Windows\System32\msiexec.exe ===
MSI (c) (28:48) [16:51:33:135]: Resetting cached policy values
MSI (c) (28:48) [16:51:33:135]: Machine policy value 'Debug' is 0
MSI (c) (28:48) [16:51:33:135]: ******* RunEngine:
           ******* Product: \GLPI-Agent-1.5-git75728309-x64.msi
           ******* Action: 
           ******* CommandLine: **********
MSI (c) (28:48) [16:51:33:137]: Client-side and UI is none or basic: Running entire install on the server.
MSI (c) (28:48) [16:51:33:137]: Grabbed execution mutex.
MSI (c) (28:48) [16:51:33:243]: Cloaking enabled.
MSI (c) (28:48) [16:51:33:243]: Attempting to enable all disabled privileges before calling Install on Server
MSI (c) (28:48) [16:51:33:248]: Incrementing counter to disable shutdown. Counter after increment: 0
MSI (s) (80:A0) [16:51:33:268]: Running installation inside multi-package transaction C:\GLPI-Agent-1.5-git75728309-x64.msi
MSI (s) (80:A0) [16:51:33:268]: Grabbed execution mutex.
MSI (s) (80:B4) [16:51:33:279]: Resetting cached policy values
MSI (s) (80:B4) [16:51:33:279]: Machine policy value 'Debug' is 0
MSI (s) (80:B4) [16:51:33:279]: ******* RunEngine:
           ******* Product: C:\GLPI-Agent-1.5-git75728309-x64.msi
           ******* Action: 
           ******* CommandLine: **********
MSI (s) (80:B4) [16:51:33:281]: Note: 1: 2203 2: C:\GLPI-Agent-1.5-git75728309-x64.msi 3: -2147287038 
MSI (s) (80:B4) [16:51:33:281]: MainEngineThread is returning 2
MSI (s) (80:A0) [16:51:33:283]: User policy value 'DisableRollback' is 0
MSI (s) (80:A0) [16:51:33:283]: Machine policy value 'DisableRollback' is 0
MSI (s) (80:A0) [16:51:33:284]: Incrementing counter to disable shutdown. Counter after increment: 0
MSI (s) (80:A0) [16:51:33:284]: Note: 1: 1402 2: HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Installer\Rollback\Scripts 3: 2 
MSI (s) (80:A0) [16:51:33:284]: Note: 1: 1402 2: HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Installer\Rollback\Scripts 3: 2 
MSI (s) (80:A0) [16:51:33:285]: Decrementing counter to disable shutdown. If counter >= 0, shutdown will be denied.  Counter after decrement: -1
MSI (c) (28:48) [16:51:33:292]: Decrementing counter to disable shutdown. If counter >= 0, shutdown will be denied.  Counter after decrement: -1
MSI (c) (28:48) [16:51:33:292]: MainEngineThread is returning 2
=== Verbose logging stopped: 02/02/2023  16:51:33 ===
```

don't have the .msi file on c:\ (maybe normal if deleted after)
